apollo-federation/src/schema/position.rs

```diff
@@ -3339,14 +3368,20 @@ impl InterfaceTypeDefinitionPosition {
         for field in &referencers.interface_fields {
             field.remove(schema)?;
         }
+        self.remove_internal(schema)?;
         Ok(Some(referencers))
     }
 
+    /// Remove this interface from the schema, and recursively remove references to the interface.
     pub(crate) fn remove_recursive(
         &self,
         schema: &mut FederationSchema,
     ) -> Result<(), FederationError> {
-        let Some(referencers) = self.remove_internal(schema)? else {
+        let Some(referencers) = schema
+            .referencers
+            .interface_types
+            .swap_remove(&self.type_name)
```

Comment:
TIL `swap_remove()` is a thing. cool!

Reply:
yeah, with indexmap you must pick either shift or swap. I don't think referencers order matters so swap is faster 🤔